src/components/SelectComponents/Combobox/__tests__/Combobox.spec.tsx

```diff
@@ -17,178 +19,209 @@ describe('Combobox', () => {
 
-            wrapper.find('input').simulate('keyDown', { key: 'Escape' });
-            wrapper.update();
+            await userEvent.keyboard('[Escape]');
 
-            expect(getInputNode()).not.toBe(document.activeElement);
+            expect(getInputNode(viewContainer)).not.toBe(document.activeElement);
             // TODO: fixMe - the callback is not triggered, even though it does work in the UI
-            // expect(mockOnBlur).toHaveBeenCalled();
+            expect(mockOnBlur).toHaveBeenCalled();
         });
         // TODO: fixMe - the component doesn't get blurred, even though it does work in the UI
-        it.skip('should blur on pressing Tab button', (done) => {
-            setFocusOnInput();
-            wrapper.find('input').simulate('keyDown', { key: 'Tab' });
-            setTimeout(() => {
-                wrapper.update();
-                expect(getInputNode()).not.toBe(document.activeElement);
-                // TODO: fixMe - the callback is not triggered, even though it does work in the UI
-                // expect(mockOnBlur).toHaveBeenCalled();
-                done();
-            });
-        });
-        it('should blur on pressing ENTER button', () => {
-            setFocusOnInput();
-            wrapper.find('input').simulate('keyDown', { key: 'Enter' });
-            wrapper.update();
-
-            expect(getInputNode()).not.toBe(document.activeElement);
+        // it.skip('should blur on pressing Tab button', async (done) => {
+        //     await setFocus();
+        //     await userEvent.keyboard('[Tab]');
+        //     setTimeout(() => {
+        //         expect(getInputNode()).not.toBe(document.activeElement);
+        //         // TODO: fixMe - the callback is not triggered, even though it does work in the UI
+        //         // expect(mockOnBlur).toHaveBeenCalled();
+        //         done();
+        //     });
+        // });
+        it('should blur on pressing ENTER button', async () => {
+            await setFocus();
+            await userEvent.keyboard('[Enter]');
+
+            expect(getInputNode(viewContainer)).not.toBe(document.activeElement);
             // TODO: fixMe - the callback is not triggered, even though it does work in the UI
-            // expect(mockOnBlur).toHaveBeenCalled();
+            expect(mockOnBlur).toHaveBeenCalled();
         });
     });
     describe('callbacks', () => {
         describe('onSelectionAdd', () => {
-            it('should be called on clicking on a suggestion', () => {
-                setFocusOnInput();
+            it('should be called on clicking on a suggestion', async () => {
+                expect(view.asFragment()).toMatchSnapshot();
+                await setFocus();
                 expect(mockOnSelectionAdd).not.toHaveBeenCalled();
 
-                wrapper.find('li').first().children().simulate('click');
+                await userEvent.click(viewContainer.querySelector('li'));
```

Comment:
please select by role, or add a comment to fix it in the next ticket.

Reply:
:+1: